libs/partners/google-vertexai/tests/integration_tests/test_tools.py

```diff
@@ -0,0 +1,166 @@
+            ),
+        }
+        | prompt
+        | llm_with_tools
+        | _TestOutputParser()
+    )
+    agent_executor = AgentExecutor(agent=agent, tools=tools, verbose=True)
+
+    response = agent_executor.invoke({"input": "What is 6 raised to the 0.43 power?"})
+    print(response)
+    assert isinstance(response, dict)
+    assert response["input"] == "What is 6 raised to the 0.43 power?"
+    assert round(float(response["output"]), 3) == 2.161
+
+
+def test_stream() -> None:
+    from langchain.chains import LLMMathChain
+
+    llm = ChatVertexAI(model_name="gemini-pro")
+    math_chain = LLMMathChain.from_llm(llm=llm)
+    tools = [
+        Tool(
+            name="Calculator",
+            func=math_chain.run,
+            description="useful for when you need to answer questions about math",
+        )
+    ]
+    response = list(llm.stream("What is 6 raised to the 0.43 power?", functions=tools))
+    assert len(response) == 1
+    # for chunk in response:
+    assert isinstance(response[0], AIMessageChunk)
+    assert "function_call" in response[0].additional_kwargs
+
+
+def test_multiple_tools() -> None:
+    from langchain.agents import AgentExecutor  # type: ignore
+    from langchain.agents.format_scratchpad import (
+        format_to_openai_function_messages,  # type: ignore
+    )
+    from langchain.chains import LLMMathChain  # type: ignore
+    from langchain.utilities import GoogleSearchAPIWrapper  # type: ignore
+
+    llm = ChatVertexAI(model_name="gemini-pro", max_output_tokens=1024)
+    math_chain = LLMMathChain.from_llm(llm=llm)
+    google_search_api_key = os.environ["GOOGLE_SEARCH_API_KEY"]
+    google_cse_id = os.environ["GOOGLE_CSE_ID"]
+    search = GoogleSearchAPIWrapper(
+        k=10, google_api_key=google_search_api_key, google_cse_id=google_cse_id
+    )
+    tools = [
+        Tool(
+            name="Calculator",
+            func=math_chain.run,
+            description="useful for when you need to answer questions about math",
+        ),
+        Tool(
+            name="Search",
+            func=search.run,
+            description=(
+                "useful for when you need to answer questions about current events. "
+                "You should ask targeted questions"
+            ),
+        ),
+    ]
+    prompt = ChatPromptTemplate.from_messages(
+        [
+            ("user", "{input}"),
+            MessagesPlaceholder(variable_name="agent_scratchpad"),
+        ]
+    )
+    llm_with_tools = llm.bind(functions=tools)
+
+    agent = (
+        {  # type: ignore
+            "input": lambda x: x["input"],
+            "agent_scratchpad": lambda x: format_to_openai_function_messages(
+                x["intermediate_steps"]
+            ),
+        }
+        | prompt
+        | llm_with_tools
+        | _TestOutputParser()
+    )
+    agent_executor = AgentExecutor(agent=agent, tools=tools, verbose=True)
+
+    question = (
+        "Who is Leo DiCaprio's girlfriend? What is her "
```

Comment:
This integration test is a bit flaky for me - google search tool often returns "no good result"